sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -84,7 +84,8 @@
   private final long startEpochNanos;
   // Set of recorded attributes. DO NOT CALL any other method that changes the ordering of events.
   @GuardedBy("lock")
-  private final AttributesMap attributes;
+  @Nullable
+  private AttributesMap attributes;
```

Comment:
Why are `attributes` nullable but `links` aren't?

Reply:
Links is a final argument because the `Span` does not have an API to add links (only the builder) so it is fine to just use emptyList (which is immutable) if no links were added to avoid all null checks. In the case of attributes if I use emptyMap does not give me any benefit because that is immutable so I still need to have all checks before adding elements to the map.

Reply:
Thanks for clarifying!